src/environment.rs

```diff
@@ -214,6 +215,52 @@ pub async fn sanity_check_project(project: &Project) -> miette::Result<()> {
         );
     }
 
+    ensure_pixi_directory_and_gitignore(project).await?;
+
+    Ok(())
+}
+
+/// Ensure that the `.pixi/` directory exists and contains a `.gitignore` file.
+/// If the directory doesn't exist, create it.
+/// If the `.gitignore` file doesn't exist, create it with a '*' pattern.
+async fn ensure_pixi_directory_and_gitignore(project: &Project) -> miette::Result<()> {
+    let pixi_dir = project.pixi_dir();
+    let gitignore_path = pixi_dir.join(".gitignore");
+
+    // Create the `.pixi/` directory if it doesn't exist
+    if !pixi_dir.exists() {
+        match tokio::fs::create_dir_all(&pixi_dir).await {
+            Ok(_) => tracing::info!("Created .pixi/ directory at {}", pixi_dir.display()),
+            Err(e) => {
+                return Err(miette::miette!(
+                    "Failed to create .pixi/ directory at {}: {}",
+                    pixi_dir.display(),
+                    e
+                ))
```

Comment:
```suggestion
                return Err(e).into_diagnostic().wrap_err(format!(
                    "Failed to create .pixi/ directory at {}",
                    gitignore_path.display()
                ));
```

Wrapping the error keeps the error more readable for the user with miette.